pkg/controller/hyperconverged/hyperconverged_controller.go

```diff
@@ -121,325 +121,115 @@ func (r *ReconcileHyperConverged) Reconcile(request reconcile.Request) (reconcil
-			Labels: labels,
-		},
-	}
+func (r *ReconcileHyperConverged) ensureKubeVirtTemplateValidator(instance *hcov1alpha1.HyperConverged, logger logr.Logger, request reconcile.Request) error {
+	kubevirtTemplateValidator := newKubeVirtTemplateValidatorForCR(instance, request.Namespace)
+	return r.ensureResourceExists(instance, logger, request, kubevirtTemplateValidator)
 }
 
-func newKubevirtTemplateValidatorForCR(cr *hcov1alpha1.HyperConverged) *sspv1.KubevirtTemplateValidator {
-	labels := map[string]string{
-		"app": cr.Name,
-	}
-	return &sspv1.KubevirtTemplateValidator{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   "template-validator-" + cr.Name,
-			Labels: labels,
-		},
-	}
+func (r *ReconcileHyperConverged) ensureKWebUI(instance *hcov1alpha1.HyperConverged, logger logr.Logger, request reconcile.Request) error {
+	kWebUI := newKWebUIForCR(instance, request.Namespace)
+	return r.ensureResourceExists(instance, logger, request, kWebUI)
 }
 
-func newKWebUIForCR(cr *hcov1alpha1.HyperConverged) *kwebuis.KWebUI {
-	labels := map[string]string{
-		"app": cr.Name,
+func (r *ReconcileHyperConverged) ensureResourceExists(instance *hcov1alpha1.HyperConverged, logger logr.Logger, request reconcile.Request, desiredRuntimeObj runtime.Object) error {
+	desiredMetaObj := desiredRuntimeObj.(metav1.Object)
+
+	// use reflection to create a new default instance of desiredRuntimeObj type
+	// which is only used as a temporary variable for r.client.Get
+	typ := reflect.ValueOf(desiredRuntimeObj).Elem().Type()
+	currentRuntimeObj := reflect.New(typ).Interface().(runtime.Object)
+
+	key := client.ObjectKey{
+		Namespace: desiredMetaObj.GetNamespace(),
+		Name:      desiredMetaObj.GetName(),
 	}
-	return &kwebuis.KWebUI{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:   "kubevirt-web-ui-" + cr.Name,
-			Labels: labels,
-		},
-		// Missing CR values will be set via ENV variables of the web-ui-operator
-		Spec: kwebuis.KWebUISpec{
-			OpenshiftMasterDefaultSubdomain: cr.Spec.KWebUIMasterDefaultSubdomain, // set if provided, otherwise keep empty
-			PublicMasterHostname:            cr.Spec.KWebUIPublicMasterHostname,   // set if provided, otherwise keep empty
-			Version:                         "automatic",                          // special value to determine version dynamically from env variables; empty or missing value is reserved for deprovision
-		},
+	err := r.client.Get(context.TODO(), key, currentRuntimeObj)
+
+	if err == nil {
+		logger.Info("Skip reconcile: resource already exists", "key", key)
+		return nil
+	} else {
+		// anything other than a "not found" error, don't create the resource, simply
+		// return the error
+		if !errors.IsNotFound(err) {
+			return err
+		}
+
+		// set the reference for the object that is about to be created
+		if err = controllerutil.SetControllerReference(instance, desiredMetaObj, r.scheme); err != nil {
+			return err
+		}
+
+		// TODO: common-templates fails the Get check above and appears to still be missing.
```

Comment:
Appears so.

The CR is in the openshift ns:

````
[rwsu@localhost converged]$ kubectl get kubevirtcommontemplatesbundle -n openshift
NAME                                      AGE
common-templates-hyperconverged-cluster   4m51s
````

request.Namespace is kubevirt-hyperconverged, but key.namespace is openshift. This may be why the Get is not able to find it.

What is interesting is KWebUI which doesn't specify a namespace in the CR can be queried/Get using any namespace.

Reply:
`Kwebui`, `CDI`, and `NetworkAddons` are all cluster scoped, so that makes sense.
```
...
  names:
    kind: KWebUI
    listKind: KWebUIList
    plural: kwebuis
    singular: kwebui
  scope: Cluster
  version: v1alpha1
```

Do you think this `TODO` is no longer needed?